model-optimizer/extensions/front/CTCGreedyDecoderReplacement.py

```diff
@@ -89,3 +92,44 @@ def pattern(**kwargs):
 
     def replace_sub_graph(self, graph: Graph, match: dict):
         replace_ctc_greedy_decoder(graph, match)
+
+
+class CTCGreedyDecoderSingleReplacement(FrontReplacementSubgraph):
+    """
+    TensorFlow CTCGreedyDecoder produces output in a sparse tensor that is not supported by Inference Engine and
+    Inference Engine's CTCGreedyDecoderSeqLen has different output that is in a dense format. So this transformation
+    handle single TF CTCGreedyDecoder and warns user about another format of output.
+    """
+    enabled = True
+
+    @staticmethod
+    def pattern(**kwargs):
+        return dict(
+            nodes=[('decoder', dict(op='CTCGreedyDecoderSeqLen')),
+                   ('result', dict(op='Result')),
+                   ],
+            edges=[('decoder', 'result', {'out': 0})]
+        )
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        ctc_greedy_decoder_tf = match['decoder']
+        ctc_greedy_decoder_tf_name = ctc_greedy_decoder_tf.soft_get('name', ctc_greedy_decoder_tf.id)
+
+        # for normalizing input channel need to transpose input data from [T, N, C] to [N, T, C]
+        # which supported CTCGreedyDecoderSeqLen op.
+        log.warning('Found CTCGreedyDecoder operation at the end of network. '
+                    'PLEASE NOTE, appropriate network output will have dense format, not sparse format')
+        ctc_data_permute = create_op_with_const_inputs(graph, Transpose, {1: int64_array([1, 0, 2])},
+                                                       {'name': ctc_greedy_decoder_tf_name + '/ctc_data_permute'})
+
+        assert ctc_greedy_decoder_tf.has_valid('merge_repeated'), \
+            'The CTCGreedyDecoderSeqLen node "{}" misses "merge_repeated" attribute'.format(ctc_greedy_decoder_tf_name)
+
+        ctc_greedy_decoder_tf.in_port(0).get_source().connect(ctc_data_permute.in_port(0))
+        ctc_greedy_decoder_tf.in_port(0).disconnect()
+        ctc_data_permute.out_port(0).connect(ctc_greedy_decoder_tf.in_port(0))
+
+        if ctc_greedy_decoder_tf.out_port(1).disconnected():
+            # Create Result operation and connect in to second output if it is not connected to any  other operation
```

Comment:
connect in to second output --> connect it to the second output

Reply:
Updated.